packages/frontend-2/components/settings/workspaces/security/sso/Wrapper.vue

```diff
@@ -0,0 +1,190 @@
+            <div class="col-span-2 font-medium">Resultant claims</div>
+
+            <template v-for="(claims, scope) in scopesAndClaims" :key="scope">
+              <div class="col-span-1">{{ scope }}</div>
+              <div class="col-span-2">{{ claims }}</div>
+            </template>
+          </div>
+        </div>
+
+        <SettingsWorkspacesSecuritySsoForm
+          :workspace-slug="workspace.slug"
+          @cancel="handleCancel"
+          @submit="handleFormSubmit"
+        />
+      </div>
+    </template>
+    <SettingsWorkspacesSecuritySsoDeleteDialog
+      v-if="provider"
+      v-model:open="isDeleteDialogOpen"
+      :provider-name="provider?.name"
+      :workspace-slug="workspace.slug"
+    />
+  </section>
+</template>
+
+<script setup lang="ts">
+import type { SettingsWorkspacesSecurity_WorkspaceFragment } from '~~/lib/common/generated/gql/graphql'
+import { useWorkspaceSso } from '~/lib/workspaces/composables/management'
+import type { SsoFormValues } from '~/lib/workspaces/helpers/types'
+import type { LayoutMenuItem } from '@speckle/ui-components'
+import { HorizontalDirection } from '~~/lib/common/composables/window'
+import { EllipsisHorizontalIcon } from '@heroicons/vue/24/solid'
+
+const props = defineProps<{
+  workspace: SettingsWorkspacesSecurity_WorkspaceFragment
+}>()
+
+enum ActionTypes {
+  Delete = 'delete'
+}
+
+const apiOrigin = useApiOrigin()
+const logger = useLogger()
+const menuId = useId()
+const { provider, loading } = useWorkspaceSso({
+  workspaceSlug: props.workspace.slug
+})
+
+const isFormVisible = ref(false)
+const isEditing = ref(false)
+const showActionsMenu = ref(false)
+const isDeleteDialogOpen = ref(false)
+
+const scopesAndClaims = ref({
+  openid: '-',
+  profile: 'name, given_name, family_name',
+  email: 'email'
+})
+
+const actionsItems = computed<LayoutMenuItem[][]>(() => [
+  [{ title: 'Remove provider...', id: ActionTypes.Delete }]
+])
+
+const onActionChosen = (params: { item: LayoutMenuItem; event: MouseEvent }) => {
+  const { item } = params
+
+  switch (item.id) {
+    case ActionTypes.Delete:
+      isDeleteDialogOpen.value = true
+      break
+  }
+}
+
+const onButtonClick = () => {
+  showActionsMenu.value = !showActionsMenu.value
+}
+
+const handleConfigureClick = () => {
+  isFormVisible.value = true
+}
+
+const handleFormSubmit = (data: SsoFormValues) => {
+  logger.info('Form submitted:', data)
+  isEditing.value = false
+  isFormVisible.value = false
+}
+
+const handleCancel = () => {
+  isFormVisible.value = false
+  isEditing.value = false
+}
+
+const redirectUrl = computed(() => {
+  return `${apiOrigin}/api/v1/workspaces/${props.workspace.slug}/sso/oidc/callback?validate=true`
+})
+
+const formInitialData = computed(() => {
+  if (!isEditing.value || !provider.value) return undefined
+
+  return {
+    providerName: provider.value.name ?? '',
+    clientId: provider.value.clientId ?? '',
+    clientSecret: '••••••••••••••••••',
```

Comment:
Do we make sure this won't get saved and override the actual secret in downstream components?

Reply:
I've removed all edit functionality in this PR. It's not possible in backend yet, and I'll need to discuss with Chuck how to handle it.